### ISSUE TITLE
Join fails on Oracle9Dialect

### ISSUE THREAD

**Issue by khoshrou** [nhibernate/nhibernate-core#3113]
Hi 
I have just upgraded to 5.3.12 (it works in 5.2) and have a simple query running on an oracle database using Oracle9Dialect . Seems its creating wrong SQL.
Also for left outer joins I see old fashiond oracle SQL like:
```SQL
A = B (+) 
```
which is not ANSI standard like: 
```SQL
Left Outer Join A on A.x = B.y
```

Here is my query: 

``` CSharp
var sampleQuery = (from globalUser in Session.Query<GlobalUser>()

                                           join userRole in Session.Query<UserRole>() on globalUser.ID
                                               equals
                                               userRole.GlobalUser.ID

                                           select new
                                           {
                                               UserName = globalUser.UserName,
                                               userRole.Role.Name
                                           }
                            );

var list = sampleQuery.Select(x=>x.UserName).ToList();
```

Exception I get:

NHibernate.Exceptions.GenericADOException: could not execute query
[ select globaluser0_.USERNAME as col_0_0_ from ACUSER globaluser0_ ACUSERROLE userrole1_ where (userrole1_.ACUSERID=globaluser0_.ID) ]

``` SQL 
select globaluser0_.USERNAME as col_0_0_ from ACUSER globaluser0_ ACUSERROLE userrole1_ 
where (userrole1_.ACUSERID=globaluser0_.ID)
```
 ---> System.Data.OracleClient.OracleException: ORA-00933: SQL command not properly ended

clearly its a wrong sql generation. 
Do you have any idea what's happening here?

**Comment by bahusoid** [comment]
>I have just upgraded to 5.3.12 
Which version did you use before upgrade?

> Also for left outer joins I see old fashioned oracle SQL
What dialect are you using? Try with `Oracle10gDialect`

**Comment by khoshrou** [comment]
I was upgraded from 5.2 and using Oracle9Dialect

**Comment by bahusoid** [comment]
Indeed simple join in LINQ fails in 5.3 for legacy dialects that don't support ANSI joins. Test case: https://github.com/bahusoid/nhibernate-core/tree/gh3113Master

It should work OK with `Oracle10gDialect`

**Comment by khoshrou** [comment]
Thanks it works with Oracle10gDialect

**Comment by bahusoid** [comment]
No need to close it. We might fix it for old dialects

**Comment by khoshrou** [comment]
Sure and Thanks again

**Comment by fredericDelaporte** [comment]
I have checked the test after #2025 (fixing #981), which was looking as the "ideal" suspect to me, but that does still work at that point. The "right" table has a `FROM_FRAGMENT` `Type` [there](https://github.com/nhibernate/nhibernate-core/blob/d3f132346b6af0a88b8e5e89ea03f9137126f444/src/NHibernate/Hql/Ast/ANTLR/SqlGenerator.cs#L213), which avoids the issue.

But at some later commit something causes it to have an `ENTITY_JOIN` type, with which the issue occurs.

I have not analyzed yet if that is a bug to have it typed `ENTITY_JOIN` or if it should anyway be able to handle theta style join in this case too.

The simplification brought by #3116 would not solve the trouble, since the JoinSequence is not flagged as being theta style. (Seen while debugging, on #981 commit, and on current master.)

We may fix this by also checking for `ENTITY_JOIN` [here](https://github.com/nhibernate/nhibernate-core/blob/e6b8dbc4b0c16c70521e42d4b6af024354a7c1bc/src/NHibernate/Hql/Ast/ANTLR/Util/JoinProcessor.cs#L156-L161).

It should work in 5.3.x too.

**Comment by bahusoid** [comment]
> But at some later commit something causes it to have an ENTITY_JOIN

It's caused by https://github.com/nhibernate/nhibernate-core/pull/2327

**Comment by bahusoid** [comment]
Also I have a fix in mind how to  add entity join support for legacy dialects. But it seems legacy dialects are broken in many others places. See errors for Oracle job in  https://github.com/bahusoid/nhibernate-core/compare/oracle9Dialect (where all tests are executed for legacy dialect)

**Comment by fredericDelaporte** [comment]
Yes it is broken in many other tests, but it was already broken in 5.2.x for the few other places I have checked. We should focus on fixing the regression.

**Comment by fredericDelaporte** [comment]
Fixed by #3137
